packages/core/src/components/ozon-content/ozon-content.tsx

```diff
@@ -1,48 +1,199 @@
+            {this.childNodesToJsx(node)}
+          </div>
+        );
+      case "#text":
+        return <>{node.textContent}</>;
+      case "#document":
+        return <>{this.childNodesToJsx(node)}</>;
+      case "ContainerBlocksType":
+        return <div class={"od-Inhoud"}>{this.childNodesToJsx(node)}</div>;
+      case "BlockMixedcontentMetMaximaleInlinesMarkersPopupsType":
+        return <div class={"od-Opschrift"}>{this.childNodesToJsx(node)}</div>;
+      case "IntRef":
+        const intRefHref = `#${this.getAttribute(node, "ref")}`;
+        return (
+          <a href={intRefHref} onClick={this.onAnchorClick}>
+            {this.childNodesToJsx(node)}
+          </a>
+        );
+      case "ExtRef":
+        return (
+          <a target="_blank" href={this.getAttribute(node, "ref")}>
+            {this.childNodesToJsx(node)}
+          </a>
+        );
+      case "ExtIoRef":
+        return (
+          <a target="_blank" href={this.getAttribute(node, "ref")}>
+            {this.childNodesToJsx(node)}
+          </a>
+        );
+      case "Al":
+        return <p>{this.childNodesToJsx(node)}</p>;
+      case "sub":
+        return <sub>{this.childNodesToJsx(node)}</sub>;
+      case "sup":
+        return <sup>{this.childNodesToJsx(node)}</sup>;
+      case "strong":
+        return <strong>{this.childNodesToJsx(node)}</strong>;
+      case "b":
+        return <b>{this.childNodesToJsx(node)}</b>;
+      case "i":
+        return <i>{this.childNodesToJsx(node)}</i>;
+      case "u":
+        return <u>{this.childNodesToJsx(node)}</u>;
+      case "Illustratie":
+        return (
+          <img
+            src={this.getAttribute(node, "naam")}
+            alt={this.getAttribute(node, "naam")}
+            height={this.getAttribute(node, "hoogte")}
+            width={this.getAttribute(node, "breedte")}
+          />
+        );
+      case "InlineTekstAfbeelding":
+        return (
+          <img
+            src={this.getAttribute(node, "naam")}
+            alt={this.getAttribute(node, "naam")}
+            height={this.getAttribute(node, "hoogte")}
+            width={this.getAttribute(node, "breedte")}
+          />
+        );
+      case "br":
+        () => <br />;
+      case "NootNummer":
+        return <>{node.textContent}</>;
+      case "Noot":
+        const nootId = uuid();
+        const nootTermId = `dso-ozon-term-${nootId}`;
+        const nootContentId = `dso-ozon-content-${nootId}`;
+        const nootHref = `#${nootTermId}`;
+        return [
+          <a
+            class="noot"
+            href={nootHref}
+            id={nootTermId}
+            aria-controls={nootContentId}
+            aria-expanded="false"
+            onClick={this.onClickNoot}
+          >
+            <sup>{this.childNodesToJsx(node, "NootNummer")}</sup>
+          </a>,
+          <div class="od-Al" id={nootContentId}>
+            {this.childNodesToJsx(node, "Al")}
+          </div>,
+        ];
+      case "table":
```

Comment:
De cases `table`, `tgroup`, `title`, `thead`, `tbody`, `row` en `entry` hebben allemaal met het tabel component te maken. De structuur in de data is bekend en consequent.

Dit soort "data componenten" kunnen als één geheel gemapt worden naar "JSX componenten":

```xml
<table
  colsep="0"
  eId="chp_4__subsec_4.11__art_4.189__para_2__table_o_1"
  frame="topbot"
  pgwide="1"
  rowsep="1"
  tabstyle="xml2"
  wId="mnre1034_1-0__chp_4__subsec_4.11__art_4.189__para_2__table_o_1"
> <!-- <table> -->
  <title>Tabel 4.189 Emissiegrenswaarden</title> <!-- <caption> -->
  <tgroup align="left" cols="7">
    <colspec colname="col1" colnum="1" colwidth="10.7*" />
    <colspec colname="col2" colnum="2" colwidth="14.1*" />
    <colspec colname="col3" colnum="3" colwidth="11.4*" />
    <colspec colname="col4" colnum="4" colwidth="18.4*" />
    <colspec colname="col5" colnum="5" colwidth="14.6*" />
    <colspec colname="col6" colnum="6" colwidth="11.9*" />
    <colspec colname="col7" colnum="7" colwidth="18.9*" />
    <!-- Bovenstaande <colspec> nodes kunnen misschien gebruikt worden om het HTML5 element <col> te maken, of te parsen en verwerken tot variablen waar verderop mee kan worden gerekend -->
    <thead valign="bottom"> <!-- <thead> -->
      <row rowsep="1"> <!-- <tr> -->
        <entry
          align="left"
          colname="col1"
          colsep="0"
          morerows="2"
          rotate="0"
          valign="top"
        > <!-- <td> Met colname kan ongetwijfeld iets gedaan worden met de bovenstaande <colspec> nodes -->
          <!-- Inhoud die in Al <entry> mag -->
        </entry>
      </row>
    </thead>
    <tbody valign="top"> <!-- <tbody> -->
      <row rowsep="1"> <!-- <tr> -->
        <entry align="left" colname="col1" colsep="0" rotate="0" valign="top"> <!-- <td> -->
          <!-- Inhoud die in Al <entry> mag -->
        </entry>
      </row>
    </tbody>
  </tgroup>
</table>
```

Door de subnodes van de `table` node niet als losse nodes te behandelen maar de integreren in één transformatie heb je op elk niveau (van node `table` tot node `entry` en alles daartussen) alle data die je nodig hebt om de tabel te genereren.

Het lijkt mij handig dat je eerst een data validator/mapper maakt zodat je een object hebt waaruit je JSX kan genereren.

Willen we het maken van stories/content gelijktrekken aan de data naar JSX mapping? Afgezien van overduidelijke `#text` en `#document` matches.